jvm/core/src/main/scala/org/scalatest/prop/Generator.scala

```diff
@@ -2216,9 +2258,23 @@ object Generator {
             (NextRoseTree(nonZeroInt), Nil, rnd2)
         }
       }
+      override def canonicals(rnd: Randomizer): (LazyListOrStream[RoseTree[NonZeroInt]], Randomizer) = {
+        case class CanonicalRoseTree(value: NonZeroInt) extends RoseTree[NonZeroInt] {
+          def shrinks(rndPassedToShrinks: Randomizer): (LazyListOrStream[RoseTree[NonZeroInt]], Randomizer) = {
+            def resLazyList(theValue: NonZeroInt): LazyListOrStream[RoseTree[NonZeroInt]] = {
+              if (theValue.value == 1 || theValue.value == -1) LazyListOrStream.empty
+              else {
```

Comment:
@cheeseng I think if we check for both 1 and -1, we'll end up stopping at -1 every time, and we want to go to 1.